app/Models/Dataset.php

```diff
@@ -478,9 +478,37 @@ public function json()
             'features' => $features,
             'display' => $featureCollectionConfig->toArray(),
         );
+      
+        if( count($features) == 0){
+            $dataset->description = null; 
+            $dataset->warning .= "<p>0 results found</p>";

```

Comment:
Should get rid of these mutations on the model properties. If you need to make modifications on the configurations, do it after the `$data['display']` is set.

Reply:
Updated